api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionEditPaths.scala

```diff
@@ -92,38 +94,68 @@ trait SubmissionEditPaths
             }
           }
         }
-      } ~ timedPost { uri =>
-        if (editType == "macro") {
-          entity(as[MacroEditJustificationWithName]) { justifyEdit =>
-            completeVerified(institutionId, period, seqNr, uri) {
-              val supervisor = system.actorSelection("/user/supervisor")
-              val submissionId = SubmissionId(institutionId, period, seqNr)
-              val fHmdaFileValidator = (supervisor ? FindProcessingActor(HmdaFileValidator.name, submissionId)).mapTo[ActorRef]
-              val fValidationState = for {
-                a <- fHmdaFileValidator
-                j <- (a ? JustifyMacroEdit(justifyEdit.edit, justifyEdit.justification)).mapTo[MacroEditJustified]
-                state <- getValidationState(institutionId, period, seqNr)
-              } yield state
-              completeValidationState(editType, fValidationState, uri, "")
+      }
+    }
+
+  // institutions/<institutionId>/filings/<period>/submissions/<seqNr>/edits/macro
+  def justifyMacroEditPath(institutionId: String)(implicit ec: ExecutionContext) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / "macro") { (period, seqNr) =>
+      timedPost { uri =>
+        entity(as[MacroEditJustificationWithName]) { justifyEdit =>
+          completeVerified(institutionId, period, seqNr, uri) {
+            val fValidator = fHmdaFileValidator(SubmissionId(institutionId, period, seqNr))
+
+            val fValidationState = for {
+              v <- fValidator
+              j <- (v ? JustifyMacroEdit(justifyEdit.edit, justifyEdit.justification)).mapTo[MacroEditJustified]
+              state <- getValidationState(institutionId, period, seqNr)
+            } yield state
+            completeValidationState("macro", fValidationState, uri, "")
+          }
+        }
+      }
+    }
+
+  // institutions/<institutionId>/filings/<period>/submissions/<seqNr>/edits/quality
+  def verifyQualityEditsPath(institutionId: String)(implicit ec: ExecutionContext) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / "quality") { (period, seqNr) =>
+      timedPost { uri =>
+        entity(as[QualityEditsVerification]) { verification =>
+          completeVerified(institutionId, period, seqNr, uri) {
+            val verified = verification.verified
+            val fSubmissionsActor = (supervisor ? FindSubmissions(SubmissionPersistence.name, institutionId, period)).mapTo[ActorRef]
+            val subId = SubmissionId(institutionId, period, seqNr)
+            val fValidator = fHmdaFileValidator(subId)
+
+            val fSubmissions = for {
+              va <- fValidator
+              v <- va ? VerifyQualityEdits(verified)
+              sa <- fSubmissionsActor
+              s <- (sa ? GetSubmissionById(subId)).mapTo[Submission]
+            } yield s
+
+            onComplete(fSubmissions) {
+              case Success(submission) =>
```

Comment:
You may want to check that the `VerifyQualityEdits` call returned successfully here.  Maybe return both `v` and `s` on the yield statement, and make sure they've both completed.

We did something similar [here](https://github.com/cfpb/hmda-platform/blob/master/api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionBasePaths.scala#L53)